backend/achievements/conditions.go

```diff
@@ -0,0 +1,94 @@
+package achievements
+
+import (
+	"context"
+	"github.com/bcc-code/brunstadtv/backend/common"
+	"github.com/bcc-code/brunstadtv/backend/sqlc"
+	"github.com/bcc-code/brunstadtv/backend/user"
+	"github.com/bcc-code/brunstadtv/backend/utils"
+)
+
+// Constants for string keys
+const (
+	ActionCompleted = "completed"
+
+	CollectionLessons = "lessons"
+	CollectionTasks   = "tasks"
+)
+
+type Action struct {
+	Collection string
+	Action     string
+}
+
+// CheckAchievements achieved since last check
+func CheckAchievements(ctx context.Context, queries *sqlc.Queries, loaders *common.BatchLoaders, action Action) error {
```

Comment:
Maybe the name should be something like "CheckForNewAchiviemnts"? Now it kinds sounds like you are checking the existing ones just reading the name.